common/persistence/persistence-tests/historyV2PersistenceTest.go

```diff
@@ -711,6 +716,10 @@ func (s *HistoryV2PersistenceSuite) append(branch []byte, events []*workflow.His
 	var resp *p.AppendHistoryNodesResponse
 
 	op := func() error {
+		var shardID *int
+		if s.ShardInfo != nil {
```

Comment:
If you have this code, you probably want to have a unit test to assert on the "shardID not set" error. Otherwise I don't think it is useful.

Reply:
removed